dwitter/views_v2.py

```diff
@@ -199,16 +223,31 @@ def report(self, request, pk=None):
         return Response(content, status=return_code)
 
 
-class CommentViewSet(viewsets.GenericViewSet):
+class CommentViewSet(viewsets.GenericViewSet,
+                     mixins.DestroyModelMixin):
     queryset = Comment.objects.all().select_related(
         'author',
     ).order_by('posted')
     serializer_class = DweetSerializer
 
+    def destroy(self, request, *args, **kwargs):
+        if not request.user.is_authenticated:
+            raise PermissionDenied()
+
+        comment = self.get_object()
+
+        # Only allow deletion of own dweets
+        # unless the deleter is a mod
+        if request.user == comment.author or request.user.is_staff:
+            comment.delete()
```

Comment:
Hm, I guess we don't have soft delete for comments iirc. Should we perhaps have that too?